pkg/kubelet/eviction/helpers.go

```diff
@@ -532,15 +570,24 @@ func makeSignalObservations(summaryProvider stats.SummaryProvider) (signalObserv
 	// build an evaluation context for current eviction signals
 	result := signalObservations{}
 
-	if memory := summary.Node.Memory; memory != nil && memory.AvailableBytes != nil {
-		result[SignalMemoryAvailable] = resource.NewQuantity(int64(*memory.AvailableBytes), resource.BinarySI)
+	if memory := summary.Node.Memory; memory != nil && memory.AvailableBytes != nil && machineInfo != nil {
```

Comment:
If capacity is 0, the threshold will be set to 0 (0 * percentage), which kinda behaves like a fail-safe. Did I misunderstand your point?